numpy/core/src/npymath/halffloat.c

```diff
@@ -20,96 +20,83 @@
  ********************************************************************
  */
 
+/* Force export of inline symbols. Those symbols are set inline for performance
+ * reason, but we still want them as part of the ABI.
+ */
+extern int npy_half_iszero(npy_half h);
+extern int npy_half_isnan(npy_half h);
+extern int npy_half_isinf(npy_half h);
+extern int npy_half_isfinite(npy_half h);
+extern int npy_half_signbit(npy_half h);
+extern npy_half npy_half_neg(npy_half h);
+extern npy_half npy_half_abs(npy_half h);
+extern npy_half npy_half_pos(npy_half h);
```

Comment:
All of these seem fine to move into the header.  Maybe just for the sake of it: can we add an `(since NumPy 1.24, 2022-05)` to the comment so in 5-10 years someone might decide to delete it without too much worry?

We want them as part of the ABI "for backwards compatibility".

Reply:
Comment updated, but we don't want these declarations in the header, this leads to multiple definitions. We want them once in this compilation unit.